src/pkg/build/assemblers/assembler_sif.go

```diff
@@ -129,3 +136,41 @@ func (a *SIFAssembler) Assemble(b *types.Bundle, path string) (err error) {
 
 	return
 }
+
+// changeOwner check the command being called with sudo with the environment
+// variable SUDO_COMMAND. Pattern match that for the singularity bin
+func changeOwner() (int, int, bool) {
+	r := regexp.MustCompile("(singularity)")
+	sudoCmd := os.Getenv("SUDO_COMMAND")
+	if !r.MatchString(sudoCmd) {
+		return 0, 0, false
+	}
+
+	if syscall.Getuid() == 0 && os.Getenv("SUDO_USER") == "root" {
+		return 0, 0, false
+	}
+
+	_uid := os.Getenv("SUDO_UID")
+	_gid := os.Getenv("SUDO_GID")
+	if _uid == "" || _gid == "" {
+		sylog.Warningf("Env vars SUDO_UID or SUDO_GID are not set, won't call chown over built SIF")
+
+		return 0, 0, false
+	}
+
+	uid, err := strconv.Atoi(_uid)
+	if err != nil {
+		sylog.Warningf("Error while calling strconv: %v", err)
+
+		return 0, 0, false
+	}
+	gid, err := strconv.Atoi(_gid)
+	if err != nil {
+		sylog.Warningf("Error while calling strconv : %v", err)
+
+		return 0, 0, false
+	}
+	chown := true
```

Comment:
This line doesn't serve a purpose. Just `return uid, gid, true` is okay.

Reply:
agree